### PR TITLE
Test RSpec integration: Sign in, reset password.

### PR DESCRIPTION
**Pull request by neoeno**
@munckymagik @JiggyPete waddayareckon

Probably should add a couple features for wrong email in forgot password form etc.

### CODE REVIEW

File: spec/features/principal_reset_password_spec.rb [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+  def and_they_submit_their_email
+    forgot_password_page.email_field.set @user.email
+    forgot_password_page.submit_button.click
+  end
+
+  def then_they_see_a_confirmation_message
+    message = I18n.t('devise.passwords.send_instructions')
+    expect(sign_in_page.flash_message.text).to eq message
+  end
```
Comment (munckymagik) [inline spec/features/principal_reset_password_spec.rb]: I learned we can use `expect(page.thing).to have_text('blah')`. Does that appeals to you?

Comment (neoeno) [inline spec/features/principal_reset_password_spec.rb]: yeah that seems good

File: spec/support/forgot_password_page.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+class ForgotPasswordPage < SitePrism::Page
+  set_url '/users/password/new'
+  set_url_matcher %r{/users/password/new}
+
+  element :email_field, '#user_email'
```
Comment (munckymagik) [inline spec/support/forgot_password_page.rb]: Can't use `t-*` style here? (and in the 3 other pages)

File: spec/features/principal_sign_in_spec.rb [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  def when_they_sign_in_with_incorrect_details
+    visit new_user_session_path
+    sign_in_page.email_field.set @user.email
+    sign_in_page.password_field.set 'not_a_password'
+    sign_in_page.submit_button.click
+  end
+
+  def they_have_logged_in
+    # We don't have access to `current_user` in feature
+    # tests, so we'll rely on this? Until we can test it on content.
```
Comment (munckymagik) [inline spec/features/principal_sign_in_spec.rb]: Should this be a 'TODO'?

File: spec/support/forgot_password_page.rb (hunk 2 above)
Comment (neoeno) [inline spec/support/forgot_password_page.rb]: I was under the impression that IDs or test classes are acceptable — but I've been a bit lax even in that regard. Will check opinions & fix

Comment (benbarnett) [inline spec/support/forgot_password_page.rb]: Yeah `t-` classes are preferable as it's less likely they'll be used for anything other than the test hook. IDs could be relied on elsewhere (probably js) and things can get messy quite quickly

**Comment by neoeno** [comment]
image

**Pull request closed without merging**